src/Akka.Streams.Kafka.Tests/Integration/RebalanceIntegrationTests.cs

```diff
@@ -22,110 +28,211 @@ public RebalanceIntegrationTests(ITestOutputHelper output, KafkaFixture fixture)
-                throw failed.Ex;
-        }
+        Log.Debug("Await initial partition assignment");
+        (await probe1RebalanceActor.ExpectMsgAsync<TopicPartitionsAssigned>()).Partitions.Should()
+            .BeEquivalentTo([tp0, tp1]);
+
+        Log.Debug("Read one message from probe1 with partition 1");
+        var m = await probe1.RequestNextAsync();
+
+        Log.Debug("Subscribe to the topic (without downstream demand)");
+        var probe2RebalanceActor = CreateTestProbe();
+        var probe2Subscription = Subscriptions.Topics(topic).WithRebalanceListener(probe2RebalanceActor.Ref);
+        var (control2, probe2) = KafkaConsumer.PlainSource(settings.WithClientId(ConsumerClientId2), probe2Subscription)
+            .ToMaterialized(this.SinkProbe<ConsumeResult<Null, string>>(), Keep.Both)
+            .Run(Materializer);
+
+        // All partitions should be revoked from consumer 1
+        Log.Debug("Await a revoke to consumer 1");
+        var revokedPartitions = (await probe1RebalanceActor.ExpectMsgAsync<TopicPartitionsRevoked>()).Partitions;
+        Log.Debug("Revoked partitions [{0}] from consumer 1", revokedPartitions);
+
+        // Single partition should be assigned back to both consumers
+        Log.Debug("Await assign to consumer 2");
+        var assignedPartitionConsumer2 =
+            (await probe2RebalanceActor.ExpectMsgAsync<TopicPartitionsAssigned>()).Partitions.Single();
+        Log.Debug("Assigned partition [{0}] to consumer 2", assignedPartitionConsumer2);
+
+        Log.Debug("Awaiting assign to consumer 1");
+        var assignedPartitionConsumer1 =
+            (await probe1RebalanceActor.ExpectMsgAsync<TopicPartitionsAssigned>()).Partitions.Single();
+        Log.Debug("Assigned partition [{0}] to consumer 1", assignedPartitionConsumer1);
+
+        // sanity check
+        Assert.NotEqual(assignedPartitionConsumer1, assignedPartitionConsumer2);
+
+        Log.Debug("Resume demand on both consumers");
+        _ = probe1.RequestAsync(count);
+        _ = probe2.RequestAsync(count);
+
+        // winning node is whoever was assigned to partition1
+        var (winningConsumer, losingConsumer)
+            = assignedPartitionConsumer1 == tp1 ? (probe1, probe2) : (probe2, probe1);
+
+        Log.Debug("Read all messages from winning consumer");
+        var winningMessages = await winningConsumer.ExpectNextNAsync(count).ToListAsync();
+
+        Log.Debug("Expect no messages on losing consumer");
+        await losingConsumer.ExpectNoMsgAsync(TimeSpan.FromMilliseconds(500));
+
+        Assert.Equal(count, winningMessages.Count);
+
+        await probe1.CancelAsync();
+        await probe2.CancelAsync();
+
+        // TODO: bug - if we've cancelled the stream, the controls should complete automatically
+        // Assert.True(control1.IsShutdown.IsCompleted);
+        // Assert.True(control2.IsShutdown.IsCompleted);
+    }
+
+    [Fact(Skip = "Filtering is not enabled on sub-sources just yet")]
```

Comment:
Not working yet because honestly the sub-sources are a mess and need some major work before I can even fix them correctly.